tools/makessp/makessp.py

```diff
@@ -0,0 +1,104 @@
+"""
+Copyright 2019-2020 CivicActions, Inc. See the README file at the top-level
+directory of this distribution and at https://github.com/CivicActions/compliancetools#copyright.
+"""
+from pathlib import Path
+
+import click
+import yaml
+
+from tools.helpers import ssptoolkit
+from tools.helpers.ssp import Ssp
+from tools.helpers.ssptoolkit import find_toc_tag
+from tools.makefamilies.family import Control
+from tools.makefamilies.makefamilies import create_family
+
+project = ssptoolkit.get_project()
+
+
+def get_family_data(output_to: str, family_data: dict) -> Ssp:
+    output = Path(output_to)
+    if not output.exists():
+        print(f"Creating output directory {output.as_posix()}")
+        output.mkdir(exist_ok=False)
+    families = [family for _, family in family_data.items()]
+    return Ssp(
+        name=project.name,
+        standards=project.standards,
+        families=families,
+    )
+
+
+def get_standards() -> list:
+    standards: list = []
+    for std in project.standards:
+        with open(Path(std), "r") as fp:
+            std_data = yaml.safe_load(fp)
+            standards.append(std_data.get("name"))
+    return standards
+
+
+def get_controls(control: Control) -> list:
+    control_text: list = [
+        f"#### {control.control_id}: {control.control_name}",
+        "```text",
+        control.description,
+        "```",
+        f"**Status:** {control.status}",
+    ]
+    for key, part in control.parts.items():
+        if key != "_default":
+            control_text.append(f"#### {key}")
+        for control_part in part:
+            control_text.append(f"##### {control_part.party}")
+            control_text.append(control_part.narrative)
+
+    return control_text
+
+
+def write_ssp(ssp_data: Ssp, output_to: str):
+    text_output: list = [
+        "<!--TOC-->",
+        "<!--TOC-->",
+        "",
+        f"# {project.name} System Security Plan",
+        "",
+    ]
+    standards = get_standards()
+    for standard in standards:
+        text_output.append(f"## {standard}")
+    text_output.append("\n")
+
+    for family in ssp_data.families:
```

Comment:
Couple issues just found:
1. Family order: AT and CA appear near the bottom of the TOC
2. Standard: outputs "NIST SP 800-53 Revision 5" but all controls are still "Revision 4"
3. Also, controls.md (created by makefamilies) is mode 644 but ssp.md is mode 600

Reply:
Do we have components that address PT or SR family controls?

Reply:
The file perm change is coming from the `md-toc` library. I commented out the toc call and the perms were 664, after the toc write they were 600. I'm wondering if this has always been there but we'd not noticed. I'll put in an issue with the md-toc maintainer.

Reply:
You're right that we don't have components with PT and SR families represented - yet.
Though we may want to allow choice of standard & certification as has been done on client site.

Reply:
That will be another commit. I'll approve this PR now - great work!